### PR TITLE
Hidden Workshop Blacksmithing

### PR DESCRIPTION
**Pull request by Azrun**
[feature][help wanted][input wanted]
## About the PR 
https://streamable.com/j7r6ah

Allows creating of the following items in the Hidden Workshop:
* Crafted Katana - (A weaker captains sabre compared to a typical katana.)
* Reverse Katana - When quenched in an appropriate reagent.
* Bloodthirsty Blade - When quenched in an appropriate reagent.
* Fragile Sword - When quenched when insufficient strikes have been performed on the blade.

Minimum Time to complete non-fragile sword: **25 minutes**

Bellows:
  Increases temperature of nearby Tatara.  Uses an existing sound very similar to the sound of a fuigo box bellow.

Tatara (Furnace):
  Increases temperature of unfinished blades towards their own temperature over time. (Sound and particle effect will play when blade is >=90% the furnace temperature)
  Will convert metallic rods of hardness and density >= steel into unfinished katanas over time. (LOW CHANCE)
  Slowly loses temperature.

Dojo Hammer:
  Allows you to strike an unfinished blade on an anvil that is currently hot enough to be forged (uses Action Bar to not require constant inputs similar to menders).  Chance to notify player once requisite strikes to make a katana have been performed that it can be quenched.

Unfinished Katana:
  Slowly loses heat when being worked.
  Color changes once it has been heated and as it cools. (Uses black body radiation emission model used in atmos)
  Can be quenched in a open reagenet container of volume >= a bucket when blade is hot enough to quench.
  
* Places an unfinished katana on the anvil.

## Why's this needed? 
Make the hidden workshop a destination again! Adding gating mechanism (effort & time) to re-enable Gannet's cool content.  This does allow for access to weapons but requires a very conscious effort to make them either for nefarious purposes and/or style.

## Changelog

```changelog
(u)Azrun
(*)Blades can once again be forged in the Hidden Workshop.  A lot of blood, sweat and tears must be expended to craft one.
```

### CODE REVIEW

**Comment by ChairSurfer** [comment]
this will be SO AWSOME, finally a use for this area other than some strong booze, pretty sure literally everyone thats ever gone there has tryed to make a katana, failed and left, good it will finally be a thing if this happens

**Comment by ThatFiveGuys** [comment]
This is cool but feel like 25 minutes is long to the point of being unfun. Would like to see it reduced to 15 minutes to make, considering you can get stronger weaponry in 15 minutes (plasmaglass spears, mining hammers, sec weaponry) doubt itd cause any balance issues.

**Comment by ThatFiveGuys** [comment]
Bloodthirsty katana should 100% be nerfed before this goes in, its damage increases by 5 every hit on a living mob. Pretty sure it counts non player monkeys, which means you can very easily get a weapon that deals like 120 damage. Should have a damage cap, damage increase should probably be along the lines of 3 damage and the like and maybe a cap of around 25 (Keep in mind damage resets when dropped), and should only work against alive players that arent you (or maybe instead require you to kill someone with the katana? Unsure how youd code that though)

**Comment by Eagletanker** [comment]
Just a suggestion based off nothing but the Wikipedia article, but you could rename the bloodthirsty blade Muramasa. Supposedly it’s a Japanese sword that demands blood before it can be put back into its scabbard. Also, maybe a engineering/mining traitor item could be something that creates a better sword, like a 6tc whetstone or somethkng

**Comment by blackrep** [comment]
> Just a suggestion based off nothing but the Wikipedia article, but you could rename the bloodthirsty blade Muramasa. Supposedly it’s a Japanese sword that demands blood before it can be put back into its scabbard. Also, maybe a engineering/mining traitor item could be something that creates a better sword, like a 6tc whetstone or somethkng

Pretty sure the bloodthirsty blade resembles a claymore, also has it's own naming system.
https://github.com/goonstation/goonstation/blob/da6bbc28ccff18dcba3d8f6b64d21b0657c6d2e6/code/obj/item/misc_weapons.dm#L1338

**Comment by flappybatpal** [comment]
Agree with the feedback. I think longer than 10 minutes is probably too much for weak swords, particularly if it is repetitive and involves a lot of clicking.

I think your best option is to remove bloodthirsty blade and implement an alternative, either now or in another PR.
1. It's a European claymore and not a Japanese sword.
2. It scales up to 65 brute after killing two monkeys

Magical swords could be cool but if anyone loves power gaming it tends to be people rolling scientist, although the time investment might keep them away.

**Comment by Gannets** [comment]
> 
> 
> Agree with the feedback. I think longer than 10 minutes is probably too much for average melee weapons, particularly if it is repetitive and involves a lot of clicking.
> 
> I think your best option is to remove bloodthirsty blade and look at alternatives either now or in another PR.
> 
>     1. It's a Scottish claymore and not a Japanese sword.
> 
>     2. It scales up to 65 brute after killing two monkeys
> 
> 
> Magical swords could be cool but if anyone loves power gaming it tends to be people rolling scientist, although the time investment might keep them away.

I agree with this summary. The bloodthirsty blade concept was rolled over in to the nuke ops knight class, so it doesn't really need to live here too.

I'm really pleased to see somebody working on this feature, the zone has felt unfinished since I released it..

File: buildByond.conf [hunk 1]
```diff
@@ -1,7 +1,7 @@
 BYOND_MAJOR_VERSION=514
 BYOND_MINOR_VERSION=1566
 
-SPACEMAN_DMM_VERSION=suite-1.7
+SPACEMAN_DMM_VERSION=suite-1.7.1
```
Comment (Sovexe) [inline buildByond.conf:4]: ?

Comment (Azrun) [reply buildByond.conf:4]: Addresses lint not knowing about DM particles: https://github.com/goonstation/goonstation/runs/3560370569?check_suite_focus=true

Specifically pulls in: https://github.com/SpaceManiac/SpacemanDMM/pull/275

**Review by Sovexe — CHANGES_REQUESTED** [review]

**Comment by Momoberry** [comment]
Personally agree that 12 damage and 20 stamina is not really worth the time investment when I can just pick up a fire extinguisher and be significantly more robust with that. Stamina is what will win a fight every time. I don't mind the swords having low stamina damage, but their actual damage should be at least be worth the time investment. 

Could always have the bloodthirsty steal a % of blood over having scaling damage, although i'm not sure how hard that would be. But have a % damage scaling dependant on the amount of blood stolen? scope creep tho

**Pull request merged**